src/spec/create_extension_spec.py

```diff
@@ -1,47 +1,394 @@
+                             neurodata_type_def='Testpulse',
+                             quantity='?',
+                             groups=[NWBGroupSpec(doc='Device for the testpulse data',
+                                                  neurodata_type_inc='TestpulseDevice',
+                                                  datasets=[NWBDatasetSpec(doc='Metadata about the Testpulse: Rows are the '
+                                                                           'running index, Columns are active AD channels '
+                                                                           '(up to version <= 7) or headstages (version >= 8), '
+                                                                           'the data is in the Layers.',
+                                                                           neurodata_type_inc='TestpulseMetadata'
+                                                                           ),
+                                                            NWBDatasetSpec(doc='Raw AD testpulse data',
+                                                                           neurodata_type_inc='TestpulseRawData'
+                                                                           )
+                                                            ]
+                                                  )
+                                     ]
+                             )
+
+    user_comment_device_string = NWBDatasetSpec('device specific user text notes',
+                                                neurodata_type_def='UserCommentString',
+                                                neurodata_type_inc='NWBData'
+                                                )
+
+    user_comment_device = NWBGroupSpec('Device for the user text notes',
+                                       neurodata_type_def='UserCommentDevice',
+                                       neurodata_type_inc='Device',
+                                       datasets=[NWBDatasetSpec('device specific user text notes',
+                                                                neurodata_type_inc='UserCommentString')
+                                                 ]
+                                       )
+
+    user_comment = NWBGroupSpec(doc='Free form text notes from the experimenter',
+                                neurodata_type_def='UserComment',
+                                groups=[NWBGroupSpec('Device for the user text notes',
+                                                     neurodata_type_inc='UserCommentDevice',
+                                                     datasets=[NWBDatasetSpec('device specific user text notes',
+                                                                              neurodata_type_inc='UserCommentString'
+                                                                              ),
+                                                               ]
```

Comment:
These lines are already in the NWBGroupSpec definition for the neurodata_type UserCommentDevice on line 241.
```suggestion
```